core/trie/transaction_storage.go

```diff
@@ -75,13 +85,25 @@ func (t *TransactionStorage) Get(key *bitset.BitSet) (*Node, error) {
 		return nil, err
 	}
 
+	if t.cache != nil {
```

Comment:
I think we don't need `t.cache != nil` checks in all methods if the only way we can initiate storage is by calling constructor (`NewTransactionStorage(...)`)